### PR TITLE
rework ui for copy history and style creation dialog

### PR DESCRIPTION
**Pull request by TurboGit**
After the work on the on/off history status on tooltip I found that the history&style dialogs were also impacted and the GUI was not really nice:

image

image

It is now cleaner and the code is also more versatile by setting the enabled status into the history items struct:

image

image

Would be nice to have this in darktable 3.8.1.

### CODE REVIEW

**Comment by elstoc** [comment]
When a module is disabled it might be nicer to display the on/off like in the darkroom, with off being represented by an icon in a more faded colour. Would it be clearer if the module name was also in italics (and maybe also in a faded colour) in that case?

No problem with inclusion in 3.8.1 for me, since we don't screenshot any of this stuff.

**Comment by TurboGit** [comment]
Not sure it is good to use the on/off button as now we have:

image

And one will ask with the always enabled buttons do not have the proper icon. Thinking more about this, a simple check character may be better.

**Comment by TurboGit** [comment]
So after all my best proposal is:

image

**Comment by elstoc** [comment]
Nit-picking now but it would be really nice if the "enabled" check mark was the same (angle, boldness) as the gtk one (or closer than currently).

**Comment by phweyland** [comment]
> it would be really nice if the "enabled" check mark was the same (angle, boldness) as the gtk one (or closer than currently).

Could be a the gtk togglebutton itself (disabled).

**Comment by TurboGit** [comment]
> Could be a the gtk togglebutton itself (disabled).

I tried that, but it is far too close to the toggle on first column and makes the UI confusing.

**Comment by TurboGit** [comment]
> It's too close to a letter "v" for my taste.

It is the one from dtgtk that is there since long long time... I'll try to have it a bit larger to see if it helps.

**Review by phweyland — COMMENTED** [review]
Well, for me the first solution was the best.
This said I don't know why the second version is so weird.

EDIT should be like this : image

If the check-mark is chosen I think I would prefer the widget itself.

Works for me, including the import eye. :smile:

**Comment by TurboGit** [comment]
> This said I don't know why the second version is so weird.

Because we have multiple status : on, off, always on. And in this specific UI we want only to display if the module is activated or not. We don't care if it is an always on module. So I just prefer a simple icon and no icon when not selected to match the column title which is "enabled".

**Comment by TurboGit** [comment]
A larger icon:

image

Better?

**Comment by TurboGit** [comment]
@phweyland : Also, for history module we have a standard container and the buttons on the right on your screenshot are in a vbox. For the dialog at hand here we have a treeview, and we cannot have a button/widget in a cell and not sure we can have a specific name for the icons to have the CSS displaying it enabled or not. I can investigate this a bit but I'm really no gtk expert :)

**Comment by elstoc** [comment]
After all those options... I think I prefer the first attempt

File: src/gui/hist_dialog.c [hunk 1]
```diff
@@ -244,7 +244,8 @@ int dt_gui_hist_dialog_new(dt_history_copy_item_t *d, int imgid, gboolean iscopy
   gtk_tree_selection_set_mode(gtk_tree_view_get_selection(GTK_TREE_VIEW(d->items)), GTK_SELECTION_SINGLE);
   gtk_tree_view_set_model(GTK_TREE_VIEW(d->items), GTK_TREE_MODEL(liststore));
 
-  GdkPixbuf *is_active_pb = dt_draw_paint_to_pixbuf(GTK_WIDGET(dialog), 10, dtgtk_cairo_paint_switch);
+  GdkPixbuf *is_active_pb = dt_draw_paint_to_pixbuf(GTK_WIDGET(dialog), 10, CPF_FOCUS, dtgtk_cairo_paint_switch_active);
+  GdkPixbuf *is_inactive_pb = dt_draw_paint_to_pixbuf(GTK_WIDGET(dialog), 10, 0, dtgtk_cairo_paint_switch);
```
Comment (phweyland) [inline src/gui/hist_dialog.c]: For me this one should be for active.
The inactive one should be the same dimmed.

Comment (TurboGit) [reply src/gui/hist_dialog.c]: Any idea on how to dim a cell in a treeview?

Comment (TurboGit) [reply src/gui/hist_dialog.c]: Ok, found a way:

image

Better for you?

Comment (TurboGit) [reply src/gui/hist_dialog.c]: Ok, found a way:

image

Better for you?

Comment (phweyland) [reply src/gui/hist_dialog.c]: Trying to find the way ...
Perfect for me !

**Review by phweyland — APPROVED** [review]
Good for me. Maybe just a bit too dimmed. 0.3 instead of 0.2 ?
image
EDIT: like this
image

**Comment by elstoc** [comment]
Does the enabled column _need_ the label?

**Comment by phweyland** [comment]
Would look like this:
image

**Comment by elstoc** [comment]
Hmm I'd assumed the column might be narrower. Not sure if it's better with or without.

**Comment by phweyland** [comment]
width = 30;
image

**Comment by TurboGit** [comment]
Ok, I'll remove the column label and will do the same for the style dialog.

@phweyland : `width=30` should be put where?

**Comment by TurboGit** [comment]
Never mind, found it.. So we now have:

image

and for style

image

**Comment by elstoc** [comment]
Yeah that looks good

**Comment by TurboGit** [comment]
Just found that the style edit mode dialog was missing the enabled column. Now added. Will merge if all tests are green. Thanks for the review @elstoc & @phweyland.

**Pull request merged**